security/advancedtls/crl_provider_test.go

```diff
@@ -258,6 +264,8 @@ func (s) TestFileWatcherCRLProviderDirectoryScan(t *testing.T) {
 		t.Run(tt.desc, func(t *testing.T) {
 			copyFiles(sourcePath, targetPath, tt.fileNames, t)
 			p.ScanCRLDirectory()
+			p.scanMutex.Lock()
+			defer p.scanMutex.Unlock()
```

Comment:
Anything similar here?  We really shouldn't need mutexes while testing conditions, or else they might be flaky if we lose a race.

Reply:
Here I'm not sure - I need the same instance of Provider for all the table tests, and I need to make the check `cmp.Diff(len(p.crls), tt.expectedEntries)` before the next test case scan, that's why I use mutex from `p.ScanCRLDirectory`

Reply:
Oh, I see, you're taking the mutex from the thing you're testing.

Can you call `p.CRL()` instead of directly accessing `p.crls` then?

Reply:
I think it's doable - I can provide a list of filenames to be scanned (`[]string{"2.crl", "3.crl"}` and list of CRLs to be found in the map by calling `p.CRL()` 
Some downside - the current approach ensures the map contains 'exactly x' entries (`len(p.crls)`) but the new one will be about 'at least x' entries. Is that ok?

Reply:
Oh right, `CRL()` takes a parameter and you'd need to call it multiple times.

This is OK, then, though I'd prefer stricter validation, i.e. checking the contents of the map, or at least the keys, and not just the length.

Reply:
I did the refactoring to the stricter validation and I think it looks better now (plus no more calling mutexes inside tests)